# **Disney Movies Data Mining Project**

#  **Problem Definition**

The world famous Walt Disney productions have had countless successes over the last century. Some movies, however, proved to be particularly more successful and profitable than others, classifying a dataset based on the major Disney movies released in theatres can help us figure out key elements to the success of some of these films, we want to assess their success based on their profitability.




### **Dataset Description**

For the task of news classification with machine learning, we are provided with a dataset containing Disney Movies, namely the Academy Award Review of Walt Disney Productions,United States,English,41.0,,45.472,1937-05-19,7.2,N/A, including their title, their production company, the country they were produced in, their release language, their running time, their budget and box office, their release date. It also comprises their imdb, metascore and rotten_tomatoes scores, the name of their directors and producers, their original material, their main cast, their composers,distributors, cinematographers, editors and their authors.

The Attributes are:

>* title
>* Production company
>* Country
>* Language
>* Running time
>* Budget,Box office
>* Release date
>* imdb,metascore
>* rotten_tomatoes
>* Directed by
>* Produced by
>* Based on,Starring
>* Music by
>* Distributed by
>* Cinematography
>* Edited by
>* Screenplay by

### **Tasks to be performed**


>* Importing Required Libraries
>* Analyzing the data
>* Extracting useful information and potential reasons for a success
>* Visualizing the pertinence of our findings



### **Importing Required Libraries**


In [1]:
import pandas as pd
import numpy as np

### Reading the file

In [2]:
dataset = pd.read_csv("disney_movies.csv")
dataset.head()

Unnamed: 0                            title       Production company  \
0           0          Academy Award Review of  Walt Disney Productions   
1           1  Snow White and the Seven Dwarfs  Walt Disney Productions   
2           2                        Pinocchio  Walt Disney Productions   
3           3                         Fantasia  Walt Disney Productions   
4           4             The Reluctant Dragon  Walt Disney Productions   

         Country Language  Running time     Budget    Box office Release date  \
0  United States  English          41.0        NaN  4.547200e+01   1937-05-19   
1  United States  English          83.0  1490000.0  4.180000e+08   1937-12-21   
2  United States  English          88.0  2600000.0  1.640000e+08   1940-02-07   
3  United States  English         126.0  2280000.0  8.330000e+07   1940-11-13   
4  United States  English          74.0   600000.0  9.600000e+05   1941-06-20   

   imdb  ...  rotten_tomatoes  \
0   7.2  ...              NaN   
1   7.6  ...              NaN   
2   7.4  ...              73%   
3   7.7  ...              95%   
4   6.9  ...              68%   

                                         Directed by  \
0                                                NaN   
1  ['David Hand (supervising)', 'William Cottrell...   
2  ['Ben Sharpsteen', 'Hamilton Luske', 'Bill Rob...   
3  ['Samuel Armstrong', 'James Algar', 'Bill Robe...   
4  ['Alfred Werker', '(live action)', 'Hamilton L...   

                         Produced by  \
0                                NaN   
1                        Walt Disney   
2                        Walt Disney   
3  ['Walt Disney', 'Ben Sharpsteen']   
4                        Walt Disney   

                                            Based on  \
0                                                NaN   
1         ['Snow White', 'by The', 'Brothers Grimm']   
2  ['The Adventures of Pinocchio', 'by', 'Carlo C...   
3                                                NaN   
4                                                NaN   

                                            Starring  \
0                                                NaN   
1  ['Adriana Caselotti', 'Lucille La Verne', 'Har...   
2  ['Cliff Edwards', 'Dickie Jones', 'Christian R...   
3              ['Leopold Stokowski', 'Deems Taylor']   
4  ['Robert Benchley', 'Frances Gifford', 'Buddy ...   

                                            Music by      Distributed by  \
0                                                NaN                 NaN   
1  ['Frank Churchill', 'Paul Smith', 'Leigh Harli...  RKO Radio Pictures   
2                 ['Leigh Harline', 'Paul J. Smith']  RKO Radio Pictures   
3                                        See program  RKO Radio Pictures   
4                 ['Frank Churchill', 'Larry Morey']  RKO Radio Pictures   

    Cinematography        Edited by Screenplay by  
0              NaN              NaN           NaN  
1              NaN              NaN           NaN  
2              NaN              NaN           NaN  
3  James Wong Howe              NaN           NaN  
4     Bert Giennon  Paul Weatherwax           NaN  

[5 rows x 21 columns]

## Data cleaning
In data cleaning, we check for redundant data in dataset, filter the dataset to get only important information and also rename the columns to make it convenient to process.

#### 1. Checking for redundacy

In [3]:
dataset.duplicated()

0      False
1      False
2      False
3      False
4      False
       ...  
439    False
440    False
441    False
442    False
443    False
Length: 444, dtype: bool

There is no duplicated data as shown above. 

#### 2. removing unnecessary columns from dataset
Here we remove irrelevant columns for our tasks, which is assessing movies success based on their profitability.
This columns are:
>* unnamed column
>* Production Company - Since almost all movies are produced by Walta Disney company, this column have no effect in accessing movies success
>* Language - almost all movies are in English
>* Country
>* Based on
>* Music by
>* Distributed by
>* Cinematography
>* Edited by
>* Screenplay by

All these columns are not important for our task, so we can get rid of them.

In [4]:
# Dropping unnecessary columns
dataset.drop(columns=['Unnamed: 0', 'Production company','Language','Country', 'Produced by', 'Language', 'Based on', 'Music by', 'Distributed by', 'Cinematography', 'Edited by', 'Screenplay by'], inplace=True, axis=1)


In [5]:
dataset.head()

title  Running time     Budget    Box office  \
0          Academy Award Review of          41.0        NaN  4.547200e+01   
1  Snow White and the Seven Dwarfs          83.0  1490000.0  4.180000e+08   
2                        Pinocchio          88.0  2600000.0  1.640000e+08   
3                         Fantasia         126.0  2280000.0  8.330000e+07   
4             The Reluctant Dragon          74.0   600000.0  9.600000e+05   

  Release date  imdb  metascore rotten_tomatoes  \
0   1937-05-19   7.2        NaN             NaN   
1   1937-12-21   7.6       95.0             NaN   
2   1940-02-07   7.4       99.0             73%   
3   1940-11-13   7.7       96.0             95%   
4   1941-06-20   6.9        NaN             68%   

                                         Directed by  \
0                                                NaN   
1  ['David Hand (supervising)', 'William Cottrell...   
2  ['Ben Sharpsteen', 'Hamilton Luske', 'Bill Rob...   
3  ['Samuel Armstrong', 'James Algar', 'Bill Robe...   
4  ['Alfred Werker', '(live action)', 'Hamilton L...   

                                            Starring  
0                                                NaN  
1  ['Adriana Caselotti', 'Lucille La Verne', 'Har...  
2  ['Cliff Edwards', 'Dickie Jones', 'Christian R...  
3              ['Leopold Stokowski', 'Deems Taylor']  
4  ['Robert Benchley', 'Frances Gifford', 'Buddy ...

In [6]:
dataset.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 444 entries, 0 to 443
Data columns (total 10 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   title            444 non-null    object 
 1   Running time     430 non-null    float64
 2   Budget           278 non-null    float64
 3   Box office       358 non-null    float64
 4   Release date     435 non-null    object 
 5   imdb             424 non-null    float64
 6   metascore        293 non-null    float64
 7   rotten_tomatoes  391 non-null    object 
 8   Directed by      443 non-null    object 
 9   Starring         409 non-null    object 
dtypes: float64(5), object(5)
memory usage: 34.8+ KB


#### Renaming columns for convenience.

In [7]:
dataset = dataset.rename(columns={'Running time': 'Running_time',
                       'Box office': 'Box_office',
                       'Release date': 'Release_date',
                       'Directed by': 'Directed_by'})
dataset.head()

title  Running_time     Budget    Box_office  \
0          Academy Award Review of          41.0        NaN  4.547200e+01   
1  Snow White and the Seven Dwarfs          83.0  1490000.0  4.180000e+08   
2                        Pinocchio          88.0  2600000.0  1.640000e+08   
3                         Fantasia         126.0  2280000.0  8.330000e+07   
4             The Reluctant Dragon          74.0   600000.0  9.600000e+05   

  Release_date  imdb  metascore rotten_tomatoes  \
0   1937-05-19   7.2        NaN             NaN   
1   1937-12-21   7.6       95.0             NaN   
2   1940-02-07   7.4       99.0             73%   
3   1940-11-13   7.7       96.0             95%   
4   1941-06-20   6.9        NaN             68%   

                                         Directed_by  \
0                                                NaN   
1  ['David Hand (supervising)', 'William Cottrell...   
2  ['Ben Sharpsteen', 'Hamilton Luske', 'Bill Rob...   
3  ['Samuel Armstrong', 'James Algar', 'Bill Robe...   
4  ['Alfred Werker', '(live action)', 'Hamilton L...   

                                            Starring  
0                                                NaN  
1  ['Adriana Caselotti', 'Lucille La Verne', 'Har...  
2  ['Cliff Edwards', 'Dickie Jones', 'Christian R...  
3              ['Leopold Stokowski', 'Deems Taylor']  
4  ['Robert Benchley', 'Frances Gifford', 'Buddy ...

### Extracting the two main actors.
In the starring column, there are many actors mentioned. We are only interested in main actor and the second main actor.
Some movies have multiple directors, some have only one. For those movies with multiple directors, we only need the main director, make sure all movies have one director.

In [8]:
# Extracting the main actor
main_actor = []
# dataset.isnull().head()
dataset.fillna('missing', inplace=True) # filling NA/NAN values with "missing"
for i in range(len(dataset)):
    actors = dataset.Starring[i].split(",", 2)
    main = actors[0]
    main = main.replace("'", "")
    main = main.replace('[', '')
    main = main.replace(']', '')
    main = main.replace('"', '')
    main = main.strip()
    main_actor.append(main)

dataset['Main_actor'] = main_actor

# Extracting the second actor

second_actor = []
for i in range(len(dataset)):
    actors = dataset.Starring[i].split(",", 2)
    if len(actors) == 1:
        actor2 = 'missing'
        second_actor.append(actor2)
    else:
        actor2 = actors[1]
        actor2 = actor2.replace("'", "")
        actor2 = actor2.replace('[', '')
        actor2 = actor2.replace(']', '')
        actor2 = actor2.replace('"', '')
        actor2 = actor2.strip()
        second_actor.append(actor2)
dataset['Second_actor'] = second_actor

# Extracting the main director
director = []
for i in range(len(dataset)):
    directors = dataset.Directed_by[i].split(",", 1)
    main_dir = directors[0]
    main_dir = main_dir.replace("'", "")
    main_dir = main_dir.replace('[', '')
    main_dir = main_dir.replace(']', '')
    main_dir = main_dir.replace('"', '')
    main_dir = main_dir.strip()
    director.append(main_dir)

dataset['Director'] = director

dataset.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 444 entries, 0 to 443
Data columns (total 13 columns):
 #   Column           Non-Null Count  Dtype 
---  ------           --------------  ----- 
 0   title            444 non-null    object
 1   Running_time     444 non-null    object
 2   Budget           444 non-null    object
 3   Box_office       444 non-null    object
 4   Release_date     444 non-null    object
 5   imdb             444 non-null    object
 6   metascore        444 non-null    object
 7   rotten_tomatoes  444 non-null    object
 8   Directed_by      444 non-null    object
 9   Starring         444 non-null    object
 10  Main_actor       444 non-null    object
 11  Second_actor     444 non-null    object
 12  Director         444 non-null    object
dtypes: object(13)
memory usage: 45.2+ KB


In [9]:
dataset.head()

title Running_time     Budget   Box_office  \
0          Academy Award Review of         41.0    missing       45.472   
1  Snow White and the Seven Dwarfs         83.0  1490000.0  418000000.0   
2                        Pinocchio         88.0  2600000.0  164000000.0   
3                         Fantasia        126.0  2280000.0   83300000.0   
4             The Reluctant Dragon         74.0   600000.0     960000.0   

  Release_date imdb metascore rotten_tomatoes  \
0   1937-05-19  7.2   missing         missing   
1   1937-12-21  7.6      95.0         missing   
2   1940-02-07  7.4      99.0             73%   
3   1940-11-13  7.7      96.0             95%   
4   1941-06-20  6.9   missing             68%   

                                         Directed_by  \
0                                            missing   
1  ['David Hand (supervising)', 'William Cottrell...   
2  ['Ben Sharpsteen', 'Hamilton Luske', 'Bill Rob...   
3  ['Samuel Armstrong', 'James Algar', 'Bill Robe...   
4  ['Alfred Werker', '(live action)', 'Hamilton L...   

                                            Starring         Main_actor  \
0                                            missing            missing   
1  ['Adriana Caselotti', 'Lucille La Verne', 'Har...  Adriana Caselotti   
2  ['Cliff Edwards', 'Dickie Jones', 'Christian R...      Cliff Edwards   
3              ['Leopold Stokowski', 'Deems Taylor']  Leopold Stokowski   
4  ['Robert Benchley', 'Frances Gifford', 'Buddy ...    Robert Benchley   

       Second_actor                  Director  
0           missing                   missing  
1  Lucille La Verne  David Hand (supervising)  
2      Dickie Jones            Ben Sharpsteen  
3      Deems Taylor          Samuel Armstrong  
4   Frances Gifford             Alfred Werker

#### Removing Directed_by and Starring columns
Since we have extracted our main actor, the second main actor and main director, we don't need the two columns i.e Directed_by and Starring.


In [10]:
dataset.drop(columns=['Directed_by', 'Starring'], inplace=True, axis=1)
dataset.head()

title Running_time     Budget   Box_office  \
0          Academy Award Review of         41.0    missing       45.472   
1  Snow White and the Seven Dwarfs         83.0  1490000.0  418000000.0   
2                        Pinocchio         88.0  2600000.0  164000000.0   
3                         Fantasia        126.0  2280000.0   83300000.0   
4             The Reluctant Dragon         74.0   600000.0     960000.0   

  Release_date imdb metascore rotten_tomatoes         Main_actor  \
0   1937-05-19  7.2   missing         missing            missing   
1   1937-12-21  7.6      95.0         missing  Adriana Caselotti   
2   1940-02-07  7.4      99.0             73%      Cliff Edwards   
3   1940-11-13  7.7      96.0             95%  Leopold Stokowski   
4   1941-06-20  6.9   missing             68%    Robert Benchley   

       Second_actor                  Director  
0           missing                   missing  
1  Lucille La Verne  David Hand (supervising)  
2      Dickie Jones            Ben Sharpsteen  
3      Deems Taylor          Samuel Armstrong  
4   Frances Gifford             Alfred Werker

### Removing rows with missing Data
For our task, the important columns are "Budget" and "Box_office", "imdb", "metascore", "rotten_tomatoes"
As we can see from the dataset, some rows have missing data of the above mentioned columns, so we need to get rid off such rows. 

In [11]:
# setting display.max_rows to None so that we can see the whole rows of dataset
pd.set_option('display.max_rows', None)

In [12]:
for i in range(len(dataset)):
    if(dataset['Box_office'][i]=='missing' or dataset['Budget'][i]=='missing' or dataset['imdb'][i]=='missing' or dataset['metascore'][i]=='missing' or dataset['rotten_tomatoes'][i]=='missing'):
        dataset.drop(i, inplace=True)

dataset


title Running_time  \
2                                            Pinocchio         88.0   
3                                             Fantasia        126.0   
5                                                Dumbo         64.0   
6                                                Bambi         70.0   
10                                     Make Mine Music         75.0   
11                                   Song of the South         94.0   
13                                         Melody Time         75.0   
16                                          Cinderella         74.0   
18                                 Alice in Wonderland         75.0   
20                                           Peter Pan         77.0   
22                                   The Living Desert         69.0   
25                        20,000 Leagues Under the Sea        127.0   
27                                  Lady and the Tramp         76.0   
40                                     Sleeping Beauty         75.0   
41                                      The Shaggy Dog        104.0   
49                               Swiss Family Robinson        126.0   
50                      One Hundred and One Dalmatians         79.0   
55                                    Babes in Toyland        105.0   
67                              The Sword in the Stone         79.0   
72                                        Mary Poppins        139.0   
84                                     The Jungle Book         78.0   
91                                        The Love Bug        108.0   
98                                      The Aristocats         79.0   
102                           Bedknobs and Broomsticks        118.0   
111                                         Robin Hood         83.0   
127                                      Freaky Friday         95.0   
131                                       The Rescuers         77.0   
133                                      Pete's Dragon        128.0   
141                                     The Black Hole         98.0   
147                                             Popeye        114.0   
150                                       Dragonslayer        109.0   
151                              The Fox and the Hound         83.0   
155                                                Tex        103.0   
159                                       Return to Oz        113.0   
160                                 The Black Cauldron         80.0   
163                          The Great Mouse Detective         74.0   
164                            Flight of the Navigator         90.0   
167                                   Oliver & Company         74.0   
168                           Honey, I Shrunk the Kids         93.0   
173                                         White Fang        107.0   
176                                      The Rocketeer        108.0   
177                               Beauty and the Beast         84.0   
178                                            Newsies        121.0   
179                           Honey, I Blew Up the Kid         89.0   
180                                   The Mighty Ducks        104.0   
181                                            Aladdin         90.0   
182                         The Muppet Christmas Carol         86.0   
187                                      Cool Runnings         98.0   
188                               The Three Musketeers        105.0   
190                                        Blank Check         94.0   
193                                      The Lion King         88.0   
194                             Angels in the Outfield        102.0   
196                                   The Santa Clause         97.0   
199                                   Man of the House         96.0   
202                                         Pocahontas         81.0   
203                               Operation Dumbo Drop        108.0   
20

In [13]:
dataset.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 223 entries, 2 to 443
Data columns (total 11 columns):
 #   Column           Non-Null Count  Dtype 
---  ------           --------------  ----- 
 0   title            223 non-null    object
 1   Running_time     223 non-null    object
 2   Budget           223 non-null    object
 3   Box_office       223 non-null    object
 4   Release_date     223 non-null    object
 5   imdb             223 non-null    object
 6   metascore        223 non-null    object
 7   rotten_tomatoes  223 non-null    object
 8   Main_actor       223 non-null    object
 9   Second_actor     223 non-null    object
 10  Director         223 non-null    object
dtypes: object(11)
memory usage: 29.0+ KB


After removing rows with missing data of of above mentioned columns, only 223 rows are left out of 444 rows.

### Data Transformation
###### Calculating profit and Adding category column to dataset
>* Profit = Box_office-Budget
###### The category
>* If Profit >= 2.5*Budget, then the movie is categorized as Blockbuster
>* If Budget < Profit < 2.5*Budget, then the movie is categorized as Hit
>* If Profit < Budget, then the movie is cotegorized as Flop

In [14]:
# converting the datatype of Box_office and Budget into float
dataset = dataset.astype({'Box_office':'float64', 'Budget': 'float64', 'imdb':'float64', 'metascore':'float64'})
dataset.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 223 entries, 2 to 443
Data columns (total 11 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   title            223 non-null    object 
 1   Running_time     223 non-null    object 
 2   Budget           223 non-null    float64
 3   Box_office       223 non-null    float64
 4   Release_date     223 non-null    object 
 5   imdb             223 non-null    float64
 6   metascore        223 non-null    float64
 7   rotten_tomatoes  223 non-null    object 
 8   Main_actor       223 non-null    object 
 9   Second_actor     223 non-null    object 
 10  Director         223 non-null    object 
dtypes: float64(4), object(7)
memory usage: 29.0+ KB


In [15]:
dataset.to_csv('movie_dataset.csv')
dataset = pd.read_csv('movie_dataset.csv')
category = []
dataset['Profit'] = dataset['Box_office'] - dataset['Budget']
# for i in range(len(dataset)):
#     profit.append(dataset.Box_office[i] - dataset.Budget[i])

for i in range(len(dataset)):
    if dataset.Profit[i] >= (2.5*dataset.Budget[i]):
        category.append('Blockbuster')
    elif dataset.Profit[i] > dataset.Budget[i]:
        category.append('Hit')
    else:
        category.append('Flop')

dataset['Category'] = category
dataset.drop(['Unnamed: 0','Profit'], inplace=True, axis=1)

# dataset.to_csv('movie_dataset.csv')
dataset.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 223 entries, 0 to 222
Data columns (total 12 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   title            223 non-null    object 
 1   Running_time     223 non-null    float64
 2   Budget           223 non-null    float64
 3   Box_office       223 non-null    float64
 4   Release_date     223 non-null    object 
 5   imdb             223 non-null    float64
 6   metascore        223 non-null    float64
 7   rotten_tomatoes  223 non-null    object 
 8   Main_actor       223 non-null    object 
 9   Second_actor     223 non-null    object 
 10  Director         223 non-null    object 
 11  Category         223 non-null    object 
dtypes: float64(5), object(7)
memory usage: 21.0+ KB


In [16]:
dataset.head()

title  Running_time     Budget   Box_office Release_date  imdb  \
0        Pinocchio          88.0  2600000.0  164000000.0   1940-02-07   7.4   
1         Fantasia         126.0  2280000.0   83300000.0   1940-11-13   7.7   
2            Dumbo          64.0   950000.0    1300000.0   1941-10-23   7.2   
3            Bambi          70.0   858000.0  267400000.0   1942-08-09   7.3   
4  Make Mine Music          75.0  1350000.0    3275000.0   1946-04-20   6.3   

   metascore rotten_tomatoes         Main_actor  Second_actor  \
0       99.0             73%      Cliff Edwards  Dickie Jones   
1       96.0             95%  Leopold Stokowski  Deems Taylor   
2       96.0             98%      Edward Brophy   Herman Bing   
3       91.0             90%          see below       missing   
4       60.0             70%        Nelson Eddy       missing   

               Director     Category  
0        Ben Sharpsteen  Blockbuster  
1      Samuel Armstrong  Blockbuster  
2        Ben Sharpsteen         Flop  
3  Supervising director  Blockbuster  
4           Jack Kinney          Hit

In [17]:
dataset.to_csv('final_movie_dataset.csv', index=False)